### DISCUSSION TITLE
Excluding targets for wire:loading

### DISCUSSION THREAD

**Discussion by Henk123456789** [livewire/livewire#3855] [3. Feature ideas]
Hi! This feature request has to do with the `wire:loading` element:
https://laravel-livewire.com/docs/2.x/loading-states#targeting-actions

You can target specific actions, which is great. But it would also be very handy if you could exclude targets.

I'm having a case where I always want the content of the `wire:loader` element to be shown, except for 1 action. It is a form with 10+ input targets. So if one would like to exclude the loader to be shown for 1 specific action, all 10+ targets would have to be added to the `wire:target` attribute on the `wire:loading` element. It would be great to have a `wire:excludeTarget` attribute or something like that.

I hope the feature request is clear and that other people also can benefit from it :).

**Comment by joshhanley** [comment]
@jellev123 this has been requested already in discussion #3248 and it was discussed on PR #3530.

See Caleb's comment here regarding the feature https://github.com/livewire/livewire/pull/3530#issuecomment-900622223

Are you able to provide a detailed use case with concrete examples of how you would use it in your app?

Thanks!

**Reply by Henk123456789** [reply]
Thanks for your reply! I hadn't found that discussion and PR yet.

**Use case:**
Let's say you have a component that shows a set of results, along with about 15 filters. Most filters are sliders, selectboxes or just a search field, which refresh the results after applying those filters. In these cases the contents of the element with the `wire:loading` attribute should be shown. It's all fine until now.

However, there is 1 'special' input field. For example a field where you can search for the name of a city, and according to the value that one types in, cities will be gathered through the component and get placed beneath the input field. For this I do not want the `wire:loading` element to be shown, since it is not really filtering. Only after clicking on a just received city the filter gets applied and the loader should be shown.

The HTML for the example:
```html
<div>
    <div wire:loading>
        <div class="loader">
            <div class="spinner-border text-light" role="status"></div>
        </div>
    </div>
    
    <input 
        type="text"
        id="city"
        autocomplete="off"
        wire:keydown.enter="getCities($event.target.value)"
    >

    @if(isset($filters['cities']) && is_array($filters['cities']))
    <div class="options" wire:loading.remove>
        @foreach($filters['cities'] as $id => $name)
            <button wire:click="selectCity({{ $id }})">
                {{ $name }}
            </button>
        @endforelse
    </div>
    @endif
</div>

```

The function in the component which retreives the cities
```php
public function getCities(string $search): array
{
    $this->filters['cities'] = City::where('name', 'LIKE', '%' . $search . '%')
              ->pluck('name', 'id')
              ->toArray();
}

```
<sup>I know the $search variable isn't fully user input proof here, but just for keeping the example simple</sup>

Clicking on one of the cities retreived above does actually apply the city filter, and the element with `wire:loading` should be shown just as normal. Only for retreiving the actual filter values (the cities) the loader should not be shown.

It's not a fully functional example of course, but I hope it gives an idea.

**TL;DR**
So the example here is an input which dynamically retreives filter values according to the value typed in the input, and for that I don't want to show the content of the element with the `wire:loading` attribute. Only for actually applying the filters the loader should be shown. Of course you could put all the other fields to `wire:target`, but in this case it's much cleaner to just add that one exception to let's say `wire:target.except`.

Thanks for reading!

**Reply by joshhanley** [reply]
@jellev123 yep makes perfect sense. Thanks for sharing your use case 🙂

**Reply by VincentVanWijk** [reply]
@joshhanley  I could also really use this in my current project! +1

**Comment by VesterDe** [comment]
I have another usecase, I think it's simillar but not entirely to the one above.

I have a wire:loading.class on part of a component, it works great.

On the same component, lower down, I have a wire:poll. When the poll requests fire, the other component flickers it's class very shortly every time. Docs suggest loading.delay for this, but the UX of my component becomes worse if I add the delay. I can't seem to use wire:target because the methods I would need to put in there exist only in child components, and they emitUp, and wire:target doesn't seem to pick those up, for a reason I don't understand but probably makes sense.

So, I feel like I need the opposite of wire:target, where I would put the actions triggered by my wire:poll, so that those poll requests would not trigger loading state on other components.

**Reply by joshhanley** [reply]
@VesterDe I'm not sure I follow exactly, any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it as `wire:poll` shouldn't impact any child components.

**Reply by VesterDe** [reply]
It's not a child component, it's a neighbour. Here's my code, cleaned up a bit to be shorter, if that helps:

```
<div class="white">
    <h2>Active components, currently present: <span wire:poll.5000ms=refreshActiveVoters()>{{ $ballot['active_voters'] }}</span></h2>
    @foreach ($activeComponents->toArray() as $component)
        <div class="d-block">
            {{ $component['id'] }} - <button
                wire:click="deactivateComponent('{{ $component['id'] }}')">Remove</button>
        </div>
    @endforeach
</div>

<div class="grey">
    <h2>Prepared components</h2>
    @if (count($unstartedComponents) > 0)
        <div class="max-w-screen-lg mx-auto h-full flex flex-col" wire:loading.class="opacity-70">
            @foreach ($unstartedComponents->toArray() as $component)
                <livewire:ballot-component :ballot="$ballot" :ballotComponent="$component"
                    :componentTree="$componentTree" wire:ignore
                    :wire:key="$loop->index.$loop->first.$loop->last.$component['id'].$component['editing']"
                    :first="$loop->first" :last="$loop->last" />
            @endforeach
        </div>
    @endif
</div>
```

The 5s poll calling `refreshActiveVoters` causes the `wire:loading.class` to trigger lower down. This makes sense, but I don't want it to. To prevent it, I would need to use `wire:target`, but the events I need to target are all inside that `livewire:ballot-component` component. If I specify them here, outside, `wire:target` doesn't appear to work. If making a repo would help, I can still do it.

But it would definitely be easier to say `wire:target.ignore="refreshActiveVoters"` on my component and be done with it.

**Comment by afdhalrashid** [comment]
No solution yet for this issue? 
I encounter this, but my case is quite different. I am using polling to refresh current status of event. 
In the same page, I have one dropdown(select). When I click the dropdown, the data inside it like flickering. SO, I think it is related to polling request

**Comment by c0700** [comment]
I had a similar case to exclude only few targets, so I use 2 loading nodes to implement it :
```html
<div wire:loading.remove wire:target="targetNameToExclude1, targetNameToExclude2, ..">
    <div wire:loading>
        Loading ..
    </div>
</div>
```
